sstables/sstables.cc

```diff
@@ -1939,6 +1939,28 @@ bool sstable::has_component(component_type f) const {
     return _recognized_components.contains(f);
 }
 
+std::optional<bool> sstable::originated_on_this_node() const {
+    if (_version < version_types::me) {
+        // earlier formats do not store originating host id
+        return std::nullopt;
+    }
```

Comment:
Is `me` mandatory? If not let's make it so.

Reply:
The `me` format is supported and the default since 948bc359c23c86a97f34987017edc30143c03c74 (5.1 / 2022.2)
so we can't make it make it mandatory just yet due to upgrades from 2022.1.

Reply:
https://github.com/scylladb/scylladb/issues/16551

Reply:
We can make it mandatory in master if we require users to upgrade in steps via the above releases and upgrade their sstables.

Reply:
I mean mandatory if all features are there. e.g. we don't allow the user to force md or lower.

Reply:
That's should be easy.  `md` can be taken out of the allowed options for `sstable_format` in db/config.
I don't think we need anything else.